### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by dansweet1**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,66 @@
 ![alt text](/img/logo.png "Aries Integration for Mailgun")
 # Aries Activity for Mailgun Source
 
+>An Aries integration for Mailgun's Api.
+
+### Events(https://documentation.mailgun.com/api-events.html#events)
```
Comment (trong-le) [inline README.md]: Change formatting of readme, (configurations, methods, and descriptions) to [this example] (https://github.com/aries-data/aries-activity-github-source/edit/master/README.md). [Check here for markdown formatting cheatsheet](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)

File: README.md [hunk 2]
```diff
@@ -1,5 +1,66 @@
...
+  `getReturnStats()`- returns total stats for a given domain in JSON.
+
+### Mailing List(https://documentation.mailgun.com/api-mailinglists.html#mailing-lists)
+A mailing list is a group of members (recipients) which itself has an email address. Mailgun allows the user to programmatically create mailing lists using Mailgun Mailing List API.
+  `getMailingList()`- paginate over mailing lists under your account
+
+## Example
+The following is an example in curl for a Get request of stats for ‘accepted’ and ‘delivered’ and ‘failed’ events for the past month:
+```
+curl -s --user 'api:YOUR_API_KEY' -G \
```
Comment (trong-le) [inline README.md]: Show an example response of what aries outputs rather than the curl command

File: README.md [hunk 3]
```diff
@@ -1,5 +1,66 @@
...
+
+### Stats(https://documentation.mailgun.com/api-stats.html#stats)
+Mailgun collects many different events and generates event statistics which are available in the user account's Control Panel. This data is also available via an API.
+  `getReturnStats()`- returns total stats for a given domain in JSON.
+
+### Mailing List(https://documentation.mailgun.com/api-mailinglists.html#mailing-lists)
+A mailing list is a group of members (recipients) which itself has an email address. Mailgun allows the user to programmatically create mailing lists using Mailgun Mailing List API.
+  `getMailingList()`- paginate over mailing lists under your account
+
+## Example
```
Comment (trong-le) [inline README.md]: Example Response

File: lib/index.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
+import { Activity, singleS3StreamOutput } from 'aries-data';
+import rp from 'request-promise';
+import { flattenUrlOptions } from './util.js';
+const apiUrl = 'https://api.mailgun.net/v3/';
+
+export default class MailgunSource extends Activity {
+    static props = {
```
Comment (trong-le) [inline lib/index.js]: Can remove static props

File: lib/index.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        const options = {
+            method: 'GET',
+            headers: {
+                'Authorization': `Basic ${auth}`,
+                'Content-Type': 'application/json',
+            },
+            uri: `${apiUrl}${endpoint}${uriOptions}`,
+            json: true
+        };
+        console.log(options.uri)
```
Comment (trong-le) [inline lib/index.js]: for aries logging, it's `this.log.info(options.uri);`. Also add semi colon

File: lib/index.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+const apiUrl = 'https://api.mailgun.net/v3/';
+
+export default class MailgunSource extends Activity {
+    static props = {
+        name: require('../package.json').name,
+        version: require('../package.json').version
+      };
+
+    @singleS3StreamOutput('json')
+    async onTask(activityTask, config, lastExecuted) {
```
Comment (trong-le) [inline lib/index.js]: change `lasExecuted` to `executionDate`

File: lib/index.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        const response = await rp(options);
+        const totalPages = response.pagination ? response.pagination.total_pages : 0;
+               if(totalPages <= 1)
+        return response;
+        //Key of returned object. 2 are returned. Splice "pagination".
+       let key = Object.keys(response).splice(0,1).toString();
+
+       //Start at 2 because if pages are 0 or 1 then the data has been returned
+       for(let page = 2; page <= totalPages; page++) {
+       //Update URI for next page
```
Comment (trong-le) [inline lib/index.js]: Indent whole for loop to 4 spaces

File: lib/index.js [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+       const newUriOptions = flattenUrlOptions({ page });
+       options.uri = `${apiUrl}${endpoint}${newUriOptions}`;
+       const pageData = await rp(options);
+       response[key].push(...pageData[key]);
+       }
+
+       //Since all pages are combined set total to 1
+       response.pagination.total_pages = 1;
+       return response;
+
```
Comment (trong-le) [inline lib/index.js]: remove new line

**Comment by trong-le** [comment]
@dansweet1 Also change all indentations to 4 spaces

**Comment by dansweet1** [comment]
@trong-le made suggested changes.

File: lib/index.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
+import { Activity, singleS3StreamOutput } from 'aries-data';
+import rp from 'request-promise';
+import { flattenUrlOptions } from './util.js';
+const apiUrl = 'https://api.mailgun.net/v3/';
+
+export default class MailgunSource extends Activity {
+
+    @singleS3StreamOutput('json')
+    async onTask(activityTask, config, executionDate) {
+      this.log.info(`Running with config: ${JSON.stringify(config)}`);
```
Comment (trong-le) [inline lib/index.js]: indentation to 4 spaces in this function

File: lib/index.js [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+                'Authorization': `Basic ${auth}`,
+                'Content-Type': 'application/json',
+            },
+            uri: `${apiUrl}${endpoint}${uriOptions}`,
+            json: true
+        };
+        this.log.info(options.uri);
+    const response = await rp(options);
+    const totalPages = response.pagination ? response.pagination.total_pages : 0;
+        if(totalPages <= 1)
```
Comment (trong-le) [inline lib/index.js]: fix indentation and add curly braces with return response inside of it

File: lib/index.js [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+    for(let page = 2; page <= totalPages; page++) {
+    //Update URI for next page
+        const newUriOptions = flattenUrlOptions({ page });
+        options.uri = `${apiUrl}${endpoint}${newUriOptions}`;
+        const pageData = await rp(options);
+        response[key].push(...pageData[key]);
+    }
+
+    //Since all pages are combined set total to 1
+       response.pagination.total_pages = 1;
```
Comment (trong-le) [inline lib/index.js]: should be same level where `for` is

File: lib/index.js [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+        const options = {
+            method: 'GET',
+            headers: {
+                'Authorization': `Basic ${auth}`,
+                'Content-Type': 'application/json',
+            },
+            uri: `${apiUrl}${endpoint}${uriOptions}`,
+            json: true
+        };
+        this.log.info(options.uri);
```
Comment (trong-le) [inline lib/index.js]: should be same level as response below

File: lib/index.js [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+     * @param {string} accessToken
+     * @returns {object}
+     */
+    async apiGetRequest({ endpoint, uriOptions, config }) {
+        if (!uriOptions) {
+            uriOptions = '';
+        }
+    //Variable to set up Basic HTTP Authorization
+    const auth = new Buffer(`${config.username}:${config.apiKey}`).toString('base64');
+        const options = {
```
Comment (trong-le) [inline lib/index.js]: indentation should be same level as auth above

File: lib/index.js [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+            body: body,
+            uri: `${apiUrl}${endpoint}`,
+            json: true
+        };
+        const response = await rp(options);
+        return response;
+    }
+    //Lists the connection Settings connected to the Domain on the Account
+    async getListConnectionSettings(config) {
+      const endpoint = `domains/${config.domain}/connection`;
```
Comment (trong-le) [inline lib/index.js]: this function and every function below should have indentation with 4 spaces

File: lib/util.js [hunk 15]
```diff
@@ -0,0 +1,19 @@
+const flattenUrlOptions = (options) => {
```
Comment (trong-le) [inline lib/util.js]: increase indentation to 4 spaces

File: package.json [hunk 16]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "aries-activity-mailgun-source",
+  "version": "1.0.0",
+  "description": "Mailgun",
+  "repository": "https://github.com/aries-data/aries-activity-mailgun",
```
Comment (trong-le) [inline package.json]: change repo to `aries-activity-mailgun-source`

**Comment by trong-le** [comment]
@dansweet1 also add scripts to [circle.yml file](https://github.com/aries-data/generator-aries-activity/blob/master/generators/app/templates/common/circle.yml) in example, change `<%= imageName %>` to `astronomerio/mailgun-source`

**Comment by dansweet1** [comment]
@trong-le just submitted indention changes. Again. Sorry.

**Pull request merged**